test-ts/end-to-end-specific-group-strategy.test.ts

```diff
@@ -0,0 +1,386 @@
+
+  const rewardsGroup0 = hre.ethers.BigNumber.from(parseUnits("10"));
+  const rewardsGroup1 = hre.ethers.BigNumber.from(parseUnits("20"));
+  const rewardsGroup2 = hre.ethers.BigNumber.from(parseUnits("30"));
+  const rewardsGroup5 = hre.ethers.BigNumber.from(parseUnits("10"));
+
+  let specificGroupStrategyDifferentFromActive: SignerWithAddress;
+  let specificGroupStrategySameAsActive: SignerWithAddress;
+
+  // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-explicit-any
+  before(async function (this: any) {
+    this.timeout(0); // Disable test timeout
+    await resetNetwork();
+
+    process.env = {
+      ...process.env,
+      TIME_LOCK_MIN_DELAY: "1",
+      TIME_LOCK_DELAY: "1",
+      MULTISIG_REQUIRED_CONFIRMATIONS: "1",
+      VALIDATOR_GROUPS: "",
+    };
+
+    [depositor0] = await randomSigner(parseUnits("300"));
+    [depositor1] = await randomSigner(parseUnits("300"));
+    [depositor2] = await randomSigner(parseUnits("300"));
+    [depositor3] = await randomSigner(parseUnits("300"));
+    [depositor4] = await randomSigner(parseUnits("300"));
+    [voter] = await randomSigner(parseUnits("300"));
+    const accounts = await hre.kit.contracts.getAccounts();
+    await accounts.createAccount().sendAndWaitForReceipt({
+      from: voter.address,
+    });
+    groups = [];
+    activatedGroupAddresses = [];
+    validators = [];
+    validatorAddresses = [];
+    for (let i = 0; i < 10; i++) {
+      const [group] = await randomSigner(parseUnits("11000"));
+      groups.push(group);
+      if (i < 3) {
+        activatedGroupAddresses.push(groups[i].address);
+      }
+      const [validator, validatorWallet] = await randomSigner(parseUnits("11000"));
+      validators.push(validator);
+      validatorAddresses.push(validators[i].address);
+
+      await registerValidatorGroup(groups[i]);
+      await registerValidatorAndAddToGroupMembers(groups[i], validators[i], validatorWallet);
+    }
+    await electMinimumNumberOfValidators(groups, voter);
+
+    specificGroupStrategyDifferentFromActive = groups[5];
+    specificGroupStrategySameAsActive = groups[0];
+  });
+
+  beforeEach(async () => {
+    await hre.deployments.fixture("core");
+    accountContract = await hre.ethers.getContract("Account");
+    managerContract = await hre.ethers.getContract("Manager");
+    stakedCeloContract = await hre.ethers.getContract("StakedCelo");
+    groupHealthContract = await hre.ethers.getContract("GroupHealth");
+    specificGroupStrategyContract = await hre.ethers.getContract("SpecificGroupStrategy");
+    defaultStrategy = await hre.ethers.getContract("DefaultStrategy");
+
+    multisigOwner0 = await hre.ethers.getNamedSigner("multisigOwner0");
+    await activateValidators(
+      managerContract,
+      defaultStrategy,
+      groupHealthContract,
+      multisigOwner0.address,
+      activatedGroupAddresses
+    );
+  });
+
+  it("deposit, rebalance, transfer and withdraw", async () => {
```

Comment:
could `activateAndVote` be called after initial deposit in order to lock CELO and cast votes for groups before any transfer happens?

The reasoning:

After a transfer, revokes and votes are scheduled for groups. If `activateAndVote` is called  after a transfer between two strategies, but before revoking the scheduled revokes of that transfer, wouldn't this cause `activateAndVote` to fail?

Reply:
I added activateAndVote also after initial deposit.

Yes, it is necessary to revoke everything that is scheduled for revoking before calling activateAndVote